crm-project/src/main/java/CRM/service/UserService.java

```diff
@@ -186,4 +214,14 @@ private void createDefaultSettingForNewUserInBoard(User user, Board board, Notif
         userSetting.setSetting(notificationSetting);
         board.addUserSettingToBoard(userSetting);
     }
+
+    private void createNewUserPermission(User user, Permission permission, Board board){
+        NotificationSetting notificationSetting = Validations.doesIdExists(2L, settingRepository);
+        UserPermission userPermission = new UserPermission();
+        userPermission.setId(0L);
+        userPermission.setUser(user);
+        userPermission.setPermission(permission);
+        board.addUserPermissionToBoard(userPermission);
+        createDefaultSettingForNewUserInBoard(user, board, notificationSetting);
+    }
```

Comment:
You are going to need to add all the notifications for this user, and not only 2L notification.
Also, create a constructor for userPermission that accepts id, user and permission instead of using the set function everytime

Reply:
אתה רוצה את זה דחוף זה קצת מכוער





[6:32](https://startgoogle-workspace.slack.com/archives/D044Z2L2CK1/p1671899570117239)
או תחכה לריפקטור


Ziv Hausler
  [6:34 PM](https://startgoogle-workspace.slack.com/archives/D044Z2L2CK1/p1671899670993539)
לא מעניין יופי כרגע
[6:34](https://startgoogle-workspace.slack.com/archives/D044Z2L2CK1/p1671899677127199)
שהווב יעבוד ואז אוכל לעזור לך בשרת בריפקטור